### PR TITLE
Updates to eliminate shellcheck failures - kubernetes/cluster/centos

### PR DESCRIPTION
**Pull request by phenixblue**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Mostly small updates to shell scripts to allow for passing hack/verify-shellcheck.sh and removing these files from hack/.shellcheck_failures, which should prevent regressions.

**Which issue(s) this PR fixes**:

Work towards #72956

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cluster/centos/config-build.sh [hunk 1]
```diff
@@ -31,16 +31,16 @@ ETCD_VERSION=${ETCD_VERSION:-"3.0.9"}
 # Define k8s version to use.
 K8S_VERSION=${K8S_VERSION:-"1.3.7"}
 
-DOCKER_DOWNLOAD_URL=\
+export DOCKER_DOWNLOAD_URL=\
```
Comment (eparis) [inline cluster/centos/config-build.sh]: These don't need exported as they are used by the same bash shell when this file is sourced....

File: cluster/centos/deployAddons.sh [hunk 2]
```diff
@@ -47,11 +47,11 @@ function deploy_dns {
 function deploy_dashboard {
   echo "Deploying Kubernetes Dashboard"
 
-  ${KUBECTL} apply -f ${KUBE_ROOT}/cluster/addons/dashboard/dashboard-secret.yaml
-  ${KUBECTL} apply -f ${KUBE_ROOT}/cluster/addons/dashboard/dashboard-configmap.yaml
-  ${KUBECTL} apply -f ${KUBE_ROOT}/cluster/addons/dashboard/dashboard-rbac.yaml
-  ${KUBECTL} apply -f ${KUBE_ROOT}/cluster/addons/dashboard/dashboard-controller.yaml
-  ${KUBECTL} apply -f ${KUBE_ROOT}/cluster/addons/dashboard/dashboard-service.yaml
+  ${KUBECTL} apply -f "${KUBE_ROOT}"/cluster/addons/dashboard/dashboard-secret.yaml
```
Comment (eparis) [inline cluster/centos/deployAddons.sh]: Why about use a quoting style like: `"${RELEASES_DIR}/kubernetes/server/bin/kube-controller-manager"` and here only quote the variable?

File: cluster/centos/config-build.sh (hunk 1 above)
Comment (phenixblue) [reply cluster/centos/config-build.sh]: Ok, I’ll change and add a directive to have shellcheck ignore

Comment (BenTheElder) [reply cluster/centos/config-build.sh]: please add a comment noting that we're ignoring it for this reason :+1:

File: cluster/centos/deployAddons.sh (hunk 2 above)
Comment (phenixblue) [reply cluster/centos/deployAddons.sh]: Ok, fixed in new commit

File: cluster/centos/config-build.sh (hunk 1 above)
Comment (phenixblue) [reply cluster/centos/config-build.sh]: Done!

**Review by BenTheElder — APPROVED** [review]
/lgtm
thank you!

**Pull request merged**